scripts/books_script.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+error_reporting(E_ALL ^ E_NOTICE);
+
+// check if there is at least 1 argument
+if ($argc === 1) {
+  trigger_error('you have to give at least 1 argument');
+  exit();
+} else if ($argc > 3) {
+  trigger_error('you can give at most 2 arguments');
+  exit();
+}
+//set category
+if ($argv[1] !== '') {
+  $category = $argv[1];
+} else {
+  trigger_error('you have to give at least 1 argument');
+}
+//set author
+if (isset($argv[2])) {
+  $author = $argv[2];
+} else
+  $author = "";
+
+$startIndex = 0;
+$resultArray = [];
+
+$url = createRequestURL($author, $category, $startIndex);
+$array = get($url);
+$nrItems = $array['totalItems'];
+
+while ($startIndex < $nrItems - 41) { //< $nrItems
+  $items = $array['items'];
+  $resultArray = addToArray($resultArray, $items);
+  $startIndex += 40;
+
+  $url = createRequestURL($author, $category, $startIndex);
+  $array = get($url);
+}
+
+print_r("finalArrayLength: " . count($resultArray) . "\n");
+print_r("first Item: " . $resultArray[0]['title'] . "\n");
+print_r("url: " . $url);
+print_r("nrItems: " . $nrItems . "\n");
+
+return $resultArray;
+
+/**
+ * create url.
+ *
+ * @param string $author
+ *   Author name.
+ * @param string $category
+ *   Category name.
+ * @param number $startIndex
+ *   Start index of item to get
+ *
+ * @return string
+ *   url.
+ */
+function createRequestURL($author, $category, $startIndex)
+{
+  $url = 'https://www.googleapis.com/books/v1/volumes?q=""';
+  $url .= 'inauthor:"' . $author . '"';
```

Comment:
there is a high chance to have errors if you are looking for authors with full names, e.g. "Frank Herbert", because of that white space. 
Hint: try to use urlencode on arguments, when you build the url.